Update ruby_services.md (#68)
DEFRA/dst-guides: teams/ruby_services.md

```ruby
factory :story do
  title { "My awesome story" }

>   trait :published do

Author: ```suggestion

  trait :published do
```

```ruby
factory :story do
  title { "My awesome story" }

>   

Author: ```suggestion

```

The issue we have found is that as time goes on, the required conditions of the object may change, and then there is contention as to whether the existing trait gets updated or the test code.

> We have then decided that every factory will have a base status of data. Any test that requires specific data to be there which are different from the `default` status of the object will then have to define those in a Trait.

Reviewer: ```suggestion
Every factory should include the essential data by default. If a test needs specific data that differs from the `default` status of the object, it should use a trait to define the data.
```

I thought "a base status of data" could be a bit clearer, so tried to reword it and ended up with this – what do you think?

Author: Sounds much better!

The issue we have found is that as time goes on, the required conditions of the object may change, and then there is contention as to whether the existing trait gets updated or the test code.

> We have then decided that every factory will have a base status of data. Any test that requires specific data to be there which are different from the `default` status of the object will then have to define those in a Trait. 

Author: ```suggestion
We have then decided that every factory will have a base status of data. Any test that requires specific data to be there which are different from the `default` status of the object will then have to define those in a Trait.
```

> For example, a test for the presence of a specific `title` should define a new trait with the expected title setup and test for it rather than use the base object. The idea is that basic data allow validations of the object through the different transition states but **will not** guarantee that the data will be consistent over time unless placed in a trait. 

Author: ```suggestion
For example, a test for the presence of a specific `title` should define a new trait with the expected title setup and test for it rather than use the base object. The idea is that basic data allow validations of the object through the different transition states but **will not** guarantee that the data will be consistent over time unless placed in a trait.
```

```ruby
factory :story do
  title { "My awesome story" }
  published { true }

>   in_print { true }  

Author: ```suggestion
  in_print { true }
```

>   

Author: ```suggestion

```

test_story = build(:story, :published_but_not_in_print)
```

#### Assemble - Assert - Act

We agree that tests should divide their Assert, Assemble and Act logic using empty lines. For example

> ```

Author: ```suggestion
```ruby
```
